src/functional/ffront/past_to_itir.py

```diff
@@ -240,9 +245,20 @@ def _visit_stencil_call_out_arg(
                 field.type.dims == first_field.type.dims for field in flattened
             ), "Incompatible fields in tuple: all fields must have the same dimensions."
 
+            field_slice = None
+            if isinstance(first_field, past.Subscript):
+                assert all(
+                    isinstance(field, past.Subscript) for field in flattened
+                ), "Incompatible field in tuple: either all fields or no field must be sliced."
+                assert all(
+                    concepts.eq_nonlocated(first_field.slice_, field.slice_) for field in flattened
+                ), "Incompatible field in tuple: all fields must be sliced in the same way."
+                field_slice = self._compute_field_slice(first_field)
+                first_field = first_field.value
+
```

Comment:
Is there a frontend check for the rule that multiple output args have to all be sliced the same way with a more domain-scientist oriented error message (with source location) already? If not, would it be too much work to add one?

Reply:
In #872 Till proposed to not do it, because longer term we want to remove the restriction, see https://github.com/GridTools/gt4py/pull/872#discussion_r930978998.

Reply:
I guess it's a choice if we want to be user friendly or eventually user friendly. I'd personally go with the former but I see temptation to go for the latter.